### PR TITLE
Updated YuktaMedia Analytics Adapter: updated request body with more details

### PR DESCRIPTION
**Pull request by shrikantpatwari**
## Type of change

- [x] Bugfix
- [x] Feature
- [x] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)

## Description of change

- Fixed issue with information collection related to native ads
- Updated code to better understand bid status.
- Added utm info collection mechanism
- Added local storage based session id
- Added user Id collection mechanism
- Used sendBeacon method of browser if supported
- Added pollyfill for Object.entries and Object.values if not available
- Fixed typo of spec file

- contact email of the adapter’s maintainer: info@yuktamedia.com, shrikant.patwari@yuktamedia.com
- [x] official adapter submission

https://github.com/prebid/prebid.github.io/pull/2165

### CODE REVIEW

**Comment by shrikantpatwari** [comment]
@jsnellbaker / @pm-harshad-mane can you please review changes, approve and merge.

**Comment by shrikantpatwari** [comment]
> Hi @shrikantpatwari,
> 
> Can you run `gulp test` once. I see one of our test cases from `adapterManager_spec.js` failing. Here's the screenshot for your reference.
> 
> Screenshot 2020-08-05 at 6 24 09 PM

I have fixed broken tests. Can you please review once and merge. Thanks

**Comment by Fawke** [comment]
Hi @shrikantpatwari,

I see that your test coverage is below 80%. Can you include few more unit test cases?

File: modules/yuktamediaAnalyticsAdapter.js [hunk 1]
```diff
@@ -3,134 +3,248 @@ import adapter from '../src/AnalyticsAdapter.js';
...
+  if (isNavigatorSendBeaconSupported()) {
+    window.navigator.sendBeacon(yuktamediaAnalyticsRequestUrl, JSON.stringify(data));
+  } else {
+    ajax(yuktamediaAnalyticsRequestUrl, undefined, JSON.stringify(data), { method: 'POST', contentType: 'text/plain' });
+  }
 }
 
+var yuktamediaAnalyticsAdapter = Object.assign(adapter(
+  {
+    emptyUrl,
```
Comment (Fawke) [inline modules/yuktamediaAnalyticsAdapter.js:89]: Why pass an empty url? If you are overriding the track function, I don't think you need to include this property as an argument to the adapter interface call.

Comment (shrikantpatwari) [reply modules/yuktamediaAnalyticsAdapter.js]: removed emptyUrl. Thanks for improving.

**Comment by shrikantpatwari** [comment]
I have updated code and test as per sessions, Please review and merge. Thanks.

**Comment by Fawke** [comment]
@shrikantpatwari,

Thanks for making those changes. Just two point you may need to look into.

- I noticed one thing which I wanted to point out. You are only calling the `send` function, for the `BID_WON` and `AUCTION_END` events and not calling it for other events like, `BID_REQUESTED`, `BID_RESPONSE`, `NO_BID`, `BID_TIMEOUT` and `AUCTION_END`.

If my understanding is correct, you are sending the event payload to your endpoint for only two events. Is it an oversight or by design?

- Your test coverage is still below 80%.
Screenshot 2020-08-10 at 11 53 28 AM

You can run this command, `gulp test` -> `gulp test-coverage` -> `gulp view-coverage` to generate the code coverage reports. (Make sure you're on node v12 before you run those commands)

**Comment by shrikantpatwari** [comment]
> @shrikantpatwari,
> 
> Thanks for making those changes. Just two point you may need to look into.
> 
> * I noticed one thing which I wanted to point out. You are only calling the `send` function, for the `BID_WON` and `AUCTION_END` events and not calling it for other events like, `BID_REQUESTED`, `BID_RESPONSE`, `NO_BID`, `BID_TIMEOUT` and `AUCTION_END`.
> 
> If my understanding is correct, you are sending the event payload to your endpoint for only two events. Is it an oversight or by design?

I am capturing data from all events in events variable which I finally send on Auction end event as well as when any bid is winning. So answer to question is this is by design.

> 
> * Your test coverage is still below 80%.
> 
> Screenshot 2020-08-10 at 11 53 28 AM
> 
> You can run this command, `gulp test` -> `gulp test-coverage` -> `gulp view-coverage` to generate the code coverage reports. (Make sure you're on node v12 before you run those commands)

Thanks, I have not checked before test coverage, I have made it above 80% now. Some of things can not be tested as there is no mechanism to test it.

**Comment by shrikantpatwari** [comment]
> LGTM

Thanks for merging

**Pull request merged**